doc/SONiC_OC_NTP_HLD.md

```diff
@@ -0,0 +1,952 @@
+- broadcast mode   
+        
+### 1.1.6 Scalability Requirements             
+Interface listening will not be enabled on all L3 interfaces. User can configure NTP source interfaces which are only a few.            
+Ntpd runs in one VRF context, default vrf or mgmt vrf.    
+Multiple ntp servers supported, also only a few.           
+    
+### 1.1.7 Warm Boot Requirements         
+NA
+
+## 1.2 Design Overview                 
+
+### 1.2.1 Basic Approach            
+Implement NTP support using transformer in sonic-mgmt-framework.    
+
+### 1.2.2 Container    
+The front end code change will be done in management-framework container including:   
+- XML file for the CLI   
+- Python script to handle CLI request (actioner)   
+- Jinja template to render CLI output (renderer)  
+- front-end code to support "show running-configuration" 
+- OpenConfig YANG model for NTP openconfig-system.yang and openconfig-system-ext.yang   
+- SONiC NTP model for NTP based on Redis DB schema of NTP   
+- transformer functions to    
+   * convert OpenConfig YANG model to SONiC YANG model for NTP related configurations   
+   * convert from Linux command "ntpq -p" output to OpenConfig NTP state YANG model   
+
+### 1.2.3 SAI Overview
+
+# 2 Functionality
+
+## 2.1 Target Deployment Use Cases 
+Manage/configure NTP via gNMI, REST and CLI interfaces.         
+
+## 2.2 Functional Description     
+Provide CLI, gNMI and REST supports for NTP related configurations.    
+
+## 2.3 Backend change to support new configurations     
+Provide changes in hostcfgd, ntp.conf.j2, ntp.keys.j2 and /etc/init.d/ntp.          
+SONiC click CLI enhancement if possible.      
+
+## 2.4 Behavior when Management IP Address is acquired via DHCP
+If the management IP address is acquired via DHCP, and if the NTP server option specifies the NTP server, /etc/dhcp/dhclient-exit-hooks.d/ntp script will generate the file /var/lib/ntp/ntp.conf.dhcp. This file is a copy of the default /etc/ntp.conf with a modified server list from the DHCP server. 
+NTP daemon only uses one of the 2 files, and /var/lib/ntp/ntp.conf.dhcp takes precedence over the default /etc/ntp.conf. It is the existing behavior and is out of the scope of this HLD.
+
+NTP source-interface, NTP vrf and NTP authentication discussed in the HLD are only guaranteed to take effect on the static configured NTP servers.
+For acquired NTP servers from DHCP server, NTP source-interface, NTP vrf and NTP authentication will only take effect if /var/lib/ntp/ntp.conf.dhcp is generated based on the /etc/ntp.conf with user configured NTP source-interface and NTP authentication.
+
+Applying the configured NTP source-interface, NTP vrf and NTP authentication to acquired NTP servers from the DHCP server is not a requirement for this release.
+ 
+# 3 Design    
+
+## 3.1 Overview    
+
+Enhancing the management framework backend code and transformer methods to add support for NTP.
+
+## 3.2 DB Changes

```

Comment:
Please put all the config DB schema change details in this section, even though they are spread across multiple sections in the doc. Please consolidate them into this section.